### PR TITLE
feat: add trade route

### PR DESCRIPTION
**Pull request by namgold**

### CODE REVIEW

File: packages/widgets/src/components/TradeRouting/helpers.ts [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+    return { name: '1inch', logoURL: 'https://s2.coinmarketcap.com/static/img/coins/64x64/8104.png' }
+  }
+
+  if (pool.exchange === 'paraswap') {
+    return { name: 'Paraswap', logoURL: 'https://s2.coinmarketcap.com/static/img/coins/64x64/14534.png' }
+  }
+
+  if (pool.exchange === '0x') {
+    return { name: '0x', logoURL: 'https://s2.coinmarketcap.com/static/img/coins/64x64/1896.png' }
+  }
```
Comment (viet-nv) [inline packages/widgets/src/components/TradeRouting/helpers.ts:16]: can remove this. our widget is using aggregator, not meta aggregator, then this never happen

File: packages/widgets/src/hooks/useThrottle.ts [hunk 2]
```diff
@@ -0,0 +1,9 @@
+import _ from 'lodash'
```
Comment (viet-nv) [inline packages/widgets/src/hooks/useThrottle.ts:1]: import throttle from 'lodash/throttle'

File: packages/widgets/src/components/Widget/index.tsx [hunk 3]
```diff
@@ -262,6 +280,9 @@ const Widget = ({
             onShowSource={() => setShowModal(ModalType.DEXES_SETTING)}
           />
         )
+      case ModalType.TRADE_ROUTE:
+        if (enableRoute) return <TradeRouting trade={trade} currencyIn={tokenInInfo} currencyOut={tokenOutInfo} />
+      // eslint-disable-next-line
```
Comment (viet-nv) [inline packages/widgets/src/components/Widget/index.tsx:285]: why?

Comment (viet-nv) [reply packages/widgets/src/components/Widget/index.tsx:285]: return null instead?

File: packages/widgets/src/utils/aggregationRouting.ts [hunk 4]
```diff
@@ -0,0 +1,370 @@
...
+  }
+
+  return {
+    routeSummary,
+    routerAddress: rawData.routerAddress,
+    fromMeta: rawData.fromMeta,
+  }
+}
+
+function formatRoutesV2(routes: SwapRoute[]): SwapRouteV2[] {
```
Comment (viet-nv) [inline packages/widgets/src/utils/aggregationRouting.ts:202]: why v2? where is v1?

Comment (namgold) [reply packages/widgets/src/utils/aggregationRouting.ts:202]: just copied from KS interface 😅

**Pull request closed without merging**